### DISCUSSION TITLE
Checks, log files if something going wrong

### DISCUSSION THREAD

**Discussion by reimuell** [solectrus/solectrus#2851] [Q&A]
Hello, after a successful and running installation Solectrus does not run anymore, after a boot. Obviously I made same changes. Are log files, error files available, any other method to see if Solectrus is running. Docker compose command tells me app is running, but no browser access possible anymore. thx

**Comment by ledermann** [comment]
Yes, there are log files, and it's very helpful to look into. Just uses this command:

```
docker compose logs
````

Run this command from the folder where `docker-compose.yml` is. To see just the logs from a **specific container only** you can use this command:

```
docker compose logs [container]
````

e.g.

```
docker compose logs app
````

**Comment by reimuell** [comment]
Hallo,
jetzt mal auf deutsch.
Ich bin relativ verzweifelt.

Ich habe das jetzt dreimal erfolgreich installiert, lief auch.
Forecast csv alles drin.
Reboot auch danach noch.

Wenn ich den Stecker ziehe oder Stromausfall, also kein Reboot, läuft die Anwendung nicht mehr. Alsist im Browser nicht mehr erreichbar. Rasperry läuft, Terminal Fenster geht auch.

Wenn ich docker PS richtig interpretiere

```
CONTAINER ID   IMAGE                                         COMMAND
          CREATED          STATUS                         PORTS      NAMES
d54512780a7a   ghcr.io/solectrus/solectrus:latest
"docker/entrypoint.s…"   43 minutes ago   Restarting (1) 3 seconds ago
        pi-app-1
90f130eff06a   ghcr.io/solectrus/forecast-collector:latest   "/bin/sh -c
'bundle …"   43 minutes ago   Up 3 minutes
pi-forecast-collector-1
85e85c0da5ad   ghcr.io/solectrus/senec-collector:latest      "/bin/sh -c
'bundle …"   43 minutes ago   Up 3 minutes (healthy)
pi-senec-collector-1
3b9574aa6cda   influxdb:2.7-alpine
 "/entrypoint.sh infl…"   44 minutes ago   Up 3 minutes (healthy)
 8086/tcp   pi-influxdb-1
64142b7ec5c7   redis:7-alpine
"docker-entrypoint.s…"   44 minutes ago   Up 3 minutes (healthy)
 6379/tcp   pi-redis-1
0ffe9aef08e0   postgres:16-alpine
"docker-entrypoint.s…"   44 minutes ago   Up 3 minutes (healthy)
 5432/tcp   pi-db-1
d9cf557ef7b4   containrrr/watchtower                         "/watchtower
--scope…"   44 minutes ago   Up 3 minutes (healthy)         8080/tcp
 pi-watchtower-1
```

wird Solectrus permanent wieder gestartet. Restarting.

Die log File, da bin ich mir nicht sicher ob ich den richtigen Abschnitt habe.

```
redis (172.18.0.7:6379) open
app-1  | Redis is up and running!
app-1  | influxdb (172.18.0.6:8086) open
app-1  | InfluxDB is up and running!
app-1  | db (172.18.0.8:5432) open
app-1  | PostgreSQL is up and running!
app-1  | Preparing database...
app-1  | Database is ready!
app-1  | => Booting Puma
app-1  | => Rails 7.1.2 application starting in production
app-1  | => Run `bin/rails server --help` for more startup options
app-1  | A server is already running (pid: 1, file:
/app/tmp/pids/server.pid).
app-1  | Exiting
app-1  | Starting SOLECTRUS...
app-1  | Version: v0.14.2 - 2024-01-07T19:02:01+01:00 - v0.14.2
app-1  | ----------------
app-1  | redis (172.18.0.7:6379) open
app-1  | Redis is up and running!
app-1  | influxdb (172.18.0.6:8086) open
app-1  | InfluxDB is up and running!
app-1  | db (172.18.0.8:5432) open
app-1  | PostgreSQL is up and running!
app-1  | Preparing database...
app-1  | Database is ready!
app-1  | => Booting Puma
app-1  | => Rails 7.1.2 application starting in production
app-1  | => Run `bin/rails server --help` for more startup options
app-1  | A server is already running (pid: 1, file:
/app/tmp/pids/server.pid).
```

sagt server is already running.

Auch stundenlanges Lesen von Docker Doku bringt mich nicht weiter.

Was kann ich den noch tun? Ein Stromausfall, also unkontrolliertes Absturz
des Raspberry kann es immer geben.

Läuft Solectrus auch auf einem Debian mit Benutzeroberfläche? Dieses
Filehandlig im Terminal, csv z. b., ist nicht lustig. Auch das Suchen von
Files ginge schneller.

Danke
Reiner Müller

**Reply by ledermann** [reply]
Tut mir leid, dass du so einen Huddel hast. Ich denke, das bekommen wir aber hin.

Ein Stromausfall ist natürlich ein harter Brocken. Klar kann das vorkommen, aber einem Server (und nichts anderes ist der Raspi) im laufenden Betrieb den Strom abzudrehen, würde ich nicht absichtlich machen. Aber wie dem auch sei, die entscheidende Fehlermeldung ist die hier:

```
A server is already running (pid: 1, file: /app/tmp/pids/server.pid).
```

Dieses Problem habe ich bereits [korrigiert](https://github.com/solectrus/solectrus/commit/8b691d1c8cc36a673ce97be890d88af7bcbe46b3), es gibt aber noch kein neues Release. 

Du kannst aber gerne den `develop`-Branch ausprobieren, indem du von `latest` auf `develop` wechselst. Also in der `docker-compose.yml` diese Zeile

```
image: ghcr.io/solectrus/solectrus:latest
```

... austauschen gegen ...

```
image: ghcr.io/solectrus/solectrus:develop
```

Danach `docker compose pull && docker compose up -d` ausführen. Danach wird SOLECTRUS robuster gegen solche Crashs sein. 

Bezüglich Debian: SOLECTRUS erfordert Linux und Docker. Ob du ein Server-Linux oder Desktop-Linux verwendest oder nicht, ist egal. Kannst du also machen.

Ich selber benutze übrigens [Portainer](https://www.portainer.io/), um für das ganze Docker-Zeugs eine UI zu haben. Damit kannst du dann im Browser die Konfigurationen bearbeiten und editieren. Für den CSV-Import wird das aber vermutlich nicht so viel helfen, der ist wirklich noch sehr rudimentär und wirklich nur was für Kommandozeilen-Freaks ;-) Macht man aber ja nur einmal...

**Comment by reimuell** [comment]
Hallo,
erstmal danke. Wenn man noch als alter Sack von Grossrechnern oder SAP
kommt ist diese Terminalgefricke nicht so lustig. Ich weiß ja ungefähr was
ich will, aber nicht wie.
Hat funktioniert, die Development Version kann das besser ab.

Ich bin einfach an den Netzteilstecker gestoßen und abgestürzt ist er.

Das nächste Teil ist ein Shelly entweder für die Wärmepumpe oder das noch
nicht vorhandene Balkonkraftwerk.
Ich habe mal versucht den Code zu lesen, aber da ich noch keinen Shelly
habe ist mir
nicht klar ob die Variable (Ertrag, Verbrauch) Vorzeichen oder keine hat.
Aber egal, sobald ich das habe melde ich mich nochmal.
Ich denke solche Leute wie mich, Solaranlage, Wärmepumpe,
Balkonkraftwerk(e) wird es in Zukunft mehr geben. Diese Balkonkraftwerke
kosten ja nicht mehr viel.

Und Debugging werde ich mir mal reinziehen, Visual Studio evtl. das kann ja
Ruby.

mfg
Reiner Müller



Georg Ledermann ***@***.***> schrieb am Do., 29. Feb. 2024,
16:57:

**Reply by ledermann** [reply]
Freut mich, dass es nun funktioniert 👍 

Unterstützung für Wärmepumpen ist gerade in Arbeit:
https://github.com/orgs/solectrus/discussions/2861

Unterstützung für Balkonkraftwerke wird derzeitig hier thematisiert:
https://github.com/orgs/solectrus/discussions/2621
